math/check_amicable_pair.cpp

```diff
@@ -0,0 +1,74 @@
+/**
+ * 
+ * @file
+ * \brief A C++ Program to check whether a pair of number is [amicable pair](https://en.wikipedia.org/wiki/Amicable_numbers) or not.
+ * 
+ * \details
+ * Amicable Pair are two positive integers such that sum of the proper divisor 
+ * of each number is equal to the other number.
+ * @author iamnambiar
+ */
+#include <iostream>
+#include <cassert>
+
+/**
+ * Function to calculate the sum of all the proper divisor 
+ * of an integer.
+ * @param num First number.
+ * @return Sum of the proper divisor of the number.
+ */
+int sum_of_divisor(int num) {
+    // Variable to store the sum of all proper divisors.
+    int sum = 0;
+    // Below loop condition helps to reduce Time complexity by a factor of square root of the number.
+    for (int div = 2; div * div <= num; ++div) {
+        // Check 'div' is divisor of 'num'.
+        if (num % div == 0) {
+            // If both divisor are same, add once to 'sum'
+            if (div == (num / div)) {
+                sum += div;
+            } else {
+                // If both divisor are not the same, add both to 'sum'.
+                sum += (div + (num / div));
+            }
+        }
+    }
+    return sum + 1;
+}
+
+/**
+ * Function to check whether the pair is amicable or not.
+ * @param x First number.
+ * @param y Second number.
+ * @return `true` if the pair is amicable
+ * @return `false` if the pair is not amicable
+ * is not amicable.
+ */
+bool is_amicable(int x, int y) {
+    if (sum_of_divisor(x) != y) {
+        return false;
+    }
+    return (sum_of_divisor(y) == x);
```

Comment:
This can be shortened.
```suggestion
return (sum_of_divisor(x) != y) ? false : true;
```

Reply:
I think the former looks better than the latter; simple and easy to understand.

Reply:
It can be documented, though.

Reply:
> I think the former looks better than the latter; simple and easy to understand.

It so happens that the operation is commutative on the 20+ number pairs I checked and hence, the suggestion shorthand would work.
But that is not a complete definition that asks for both the checks - sum of divisors of x should equal y and vice-versa.

The original implementation conforms to the definition.